### DISCUSSION TITLE
[Rule Suggestion] Type-safe collection rules

### DISCUSSION THREAD

**Discussion by TesteurManiak** [Floating-Dartists/fd_lints#7] [Ideas]
* `avoid_first`: Prefer type-safe operation [`firstOrNull`](https://pub.dev/documentation/collection/latest/collection/IterableExtension/firstOrNull.html)
* `avoid_last`: Prefer type-safe operation [`lastOrNull`](https://pub.dev/documentation/collection/latest/collection/IterableExtension/lastOrNull.html) 
* `avoid_single`: Prefer type-safe operation `singleOrNull`
* `avoid_element_at`: Prefer type-safe operation `elementAtOrNull`
* `avoid_first_where`: Prefer type-safe operation `firstWhereOrNull`

Those methods can be found in the [collection](https://pub.dev/packages/collection) package.

**Comment by Pierre-Monier** [comment]
Great, if you are sure that the element is in you can disable this rule on your line, but this should be the default
